CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+
+We do use the assignee feature in GitHub for issues. If you find an unassigned issue, comment asking if you can be assigned, and ideally wait for a maintainer to respond. If you find an assigned issue and you want to work on it or help out, please reach out to the assignee first.
+
+Sometimes you might get an amazing idea and start working on a huge amount of code. We love and encourage excitement like this, but we do ask that before you embarking on a giant pull request, please reach out to the community first for an initial discussion. You could [file an issue](https://github.com/kcp-dev/kcp/issues/new/choose), send a discussion to our [mailing list](https://groups.google.com/g/kcp-dev), and/or join one of our [community meetings](https://github.com/kcp-dev/kcp/issues?q=is%3Aissue+is%3Aopen+label%3Acommunity-meeting).
+
+Finally, we welcome and value all types of contributions, beyond "just code"! Other types include triaging bugs, tracking down and fixing flaky tests, improving our documentation, helping answer community questions, proposing and reviewing designs, etc.
+
+
+## Priorities & milestones
+
+We prioritize issues and features both synchronously (during community meetings) and asynchronously (Slack/GitHub conversations).
+
+We group issues together into milestones. Each milestone represents a set of new features and bug fixes that we want users to try out. We aim for each milestone to take about a month from start to finish.
+
+You can see the [current list of milestones](https://github.com/kcp-dev/kcp/milestones?direction=asc&sort=due_date&state=open) in GitHub.
+
+For a given issue or pull request, its milestone may be:
+
+- **unset/unassigned**: we haven't looked at this yet, or if we have, we aren't sure if we want to do it and it needs more community discussion
+- **assigned to a named milestone**
+- **assigned to `TBD`** - we have looked at this, decided that it is important and we eventually would like to do it, but we aren't sure exactly when
+
+
+## Coding guidelines & conventions
+
+- Always be clear about what clients or client configs target. Never use an unqualified `client`. Instead, always qualify. For example:
+    - `rootClient`
+    - `orgClient`
+    - `pclusterClient`
+    - `rootKcpClient`
+    - `orgKubeClient`
+- Configs intended for `NewClusterForConfig` (i.e. today often called "admin workspace config") should uniformely be called `clusterConfig`
+    - Note: with org workspaces, the second use for the admin workspaces will go away
+    - Note 2: sometimes we use clients for same purpose, but this can be harder to read
+- Cluster-aware clients should follow similar naming conventions:
+    - `crdClusterClient`
+    - `kcpClusterClient`
+    - `kubeClusterClient`
+- `clusterName` is a kcp term. It is **NOT** a name of a physical cluster. If we mean the latter, use `pclusterName` or similar.
```

Comment:
(No action required) Why is it important for kcp's `clusterName` to be unqualified? vs using `lclusterName` for a logical cluster and `pclusterName` for physical clusters?

Reply:
@sttts wrote these, but I'll hazard a guess: the guidance is to never use `clusterName` to mean a physical cluster. If/when unqualified, `clusterName` means a logic cluster.

I maybe want to also suggest that we never use `lcluster...` or `pcluster...` - it's too easy to wonder if that was intended, or if the leading `l`/`p` was a typo.

Reply:
`clusterName` is defined by kube's ObjectMeta. Also kcp is technically about clusters in the lcluster sense. Pcluster is something on-top. We are just not that far yet to split it appart and have the syncer in the same code base.